compiler/accelerators/snax_gemmx.py

```diff
@@ -150,4 +151,54 @@ def _generate_setup_vals(
         - a reference to the SSAValue containing the calculated field value
         """
 
-        raise NotImplementedError()
+        c0 = arith.Constant.from_int_and_width(0, 32)
+        c1 = arith.Constant.from_int_and_width(1, 32)
+        knm: list = [
+            (((cst := arith.Constant.from_int_and_width(val.data, 32)),), cst.result)
+            for val in op.stride_patterns.data[0].upper_bounds
+        ]
+
+        streamer_setup_vals = list(self._generate_streamer_setup_vals(op))
+
+        ops_to_add: list[Operation] = []
+
+        assert isinstance(generic_op := op.body.block.first_op, memref_stream.GenericOp)
+
+        if isinstance(qmac := generic_op.body.block.first_op, kernel.QMacOp):
+            # gemm
+            # bypass simd and set all related values to 0
+            bypassSIMD = c1.result  # bypass simd
+            loop_bound = c0
+            csr0 = c0
+            csr1 = c0
+            csr2 = c0
+
+            # get zero points for gemm
+            assert isinstance(qmac.zp_lhs, BlockArgument)
+            zp_a = generic_op.inputs[qmac.zp_lhs.index]
+            assert isinstance(qmac.zp_rhs, BlockArgument)
+            zp_b = generic_op.inputs[qmac.zp_rhs.index]
+
+            # bitwise and with 8b'11111111 to don't fuck up bitpacking in case of negative values
```

Comment:
```suggestion
            # bitwise and with 8b'11111111 to prevent extraneous sign extension
```

Reply:
I don't understand why this is necessary actually. You can perform explicit unsigned extension of 8-bit values with extui?

Reply:
The problem is that the values are in 32 bit format, after default linalg lowerings, so extui is not possible

Reply:
Right. Okay :) . Still would like the comment to be a bit more descriptive :)